crates/rect/src/lib.rs

```diff
@@ -85,6 +89,24 @@ pub fn systems() -> SystemGroup {
     SystemGroup::new(
         "ui/rect",
         vec![
+            ensure_has_component_with_default(pixel_line_from(), line_from()),
+            ensure_has_component_with_default(pixel_line_to(), line_to()),
+            query_mut(
+                (line_from(), line_to()),
+                (pixel_line_from().changed(), pixel_line_to().changed()),
+            )
+            .to_system(|q, world, qs, _| {
+                if let Some(world_cam) = Camera::get_active(world, main_scene(), None) {
+                    if let Some(ui_cam) = Camera::get_active(world, ui_scene(), None) {
```

Comment:
Could potentially simplify this as either
`(Some(world_cam), Some(ui_cam)) = (..., ...)`
or
`Some((world_cam, ui_cam)) = (...).zip(...)`

Reply:
Ok, just lifted this code from the physics visualization